.travis.yml

```diff
@@ -5,16 +5,14 @@ rust:
   - nightly
   - beta
   - stable
-  - 1.16.0 # currently oldest supported version, but structopt example fails to build until 1.22.1 so skip tests
+  - 1.31.0 # currently oldest supported version
 script:
-  - if [ ${TRAVIS_RUST_VERSION} = "1.16.0" ]; then
-      cargo generate-lockfile &&
-      cargo update -p cfg-if --precise 0.1.9;
+  - if [ ${TRAVIS_RUST_VERSION} = "nightly" ]; then
+      # Ideally we would like to use minimal-versions with our MSRV, but this is a nightly-only option.
+      cargo generate-lockfile -Z minimal-versions
```

Comment:
I wrote something similar to this cause of the `-Z mininal-versions` PR but this wouldn't work correctly because the steps below didn't do `--locked`, which then would require doing it to the non-nightly.

Reply:
Your version of this change look good, thanks.

Does travis give access to all versions though ? It would be better to test minimal-versions with the MSRV. Something like

```shell
   if [ ${TRAVIS_RUST_VERSION} = "1.31.0" ]; then
      cargo +nightly generate-lockfile -Z minimal-versions;
    else
      cargo generate-lockfile;
    fi
```

But I'm a travis noob, so I'll let you experiment.

Reply:
I went ahead and tried the change. Just don't cherry-pick it if travis doesn't like it.